crates/trigger/src/lib.rs

```diff
@@ -425,6 +438,13 @@ impl<Executor: TriggerExecutor> TriggerAppEngine<Executor> {
         })
     }
 
+    pub fn get_client_tls_opts(
+        &self,
+        component_id: &str,
+    ) -> Option<HashMap<Authority, ParsedClientTlsOpts>> {
+        self.client_tls_opts.get(&component_id.to_string()).cloned()
+    }
```

Comment:
Nit: does this have to clone?

```suggestion
    pub fn get_client_tls_opts(
        &self,
        component_id: &str,
    ) -> Option<&HashMap<Authority, ParsedClientTlsOpts>> {
        self.client_tls_opts.get(&component_id.to_string())
    }
```